packages/dev/core/src/AudioV2/abstractAudio/subNodes/abstractAudioSubGraph.ts

```diff
@@ -29,23 +30,56 @@ export abstract class _AbstractAudioSubGraph {
      *
      * @param name The name of the sub node
      * @param callback The function to call with the named sub node
+     *
+     * @internal
      */
-    public callOnSubNode<T extends _AbstractAudioSubNode>(name: string, callback: (node: T) => void): void {
+    public callOnSubNode<T extends _AbstractAudioSubNode>(name: AudioSubNode, callback: (node: T) => void): void {
         const node = this.getSubNode(name);
         if (node) {
             callback(node as T);
             return;
         }
 
-        const promise = this._createSubNodePromises[name] ?? this._createAndAddSubNode(name);
+        const promise = this._createSubNodePromises[name] ?? this.createAndAddSubNode(name);
 
         promise.then((node) => {
             callback(node as T);
         });
     }
 
+    /**
+     * Creates the named subnode and adds it to the sub graph.
+     *
+     * @param name The name of the sub node.
+     * @returns A promise that resolves to the created sub node.
+     *
+     * @internal
+     */
+    public createAndAddSubNode(name: AudioSubNode): Promise<_AbstractAudioSubNode> {
```

Comment:
nit - if internal (and async)
```suggestion
    public _createAndAddSubNodeAsync(name: AudioSubNode): Promise<_AbstractAudioSubNode> {
```

Reply:
Do we need the underscore prefix when the whole class is internal?

Reply:
that's a good question.... we usually do it to stay consistent, though it is less important, you are right. it's more for us in this case.

Reply:
Ok, thanks. I'm going to leave it as-is for this PR because there a bunch more of these functions that should follow the same convention and I'll do them all in a separate PR.